### ISSUE TITLE
implementation of AttentiveFP

### ISSUE THREAD

**Issue by thegodone** [aimat-lab/gcnn_keras#4]
Very interesting library for graphs. Can you implement the AttentiveFP with your GAT version ?

here the PyG version I implement : https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/nn/models/attentive_fp.py

thanks in advance,

BR
Guillaume

**Comment by PatReis** [comment]
Hallo, thank you for your comment.
the repo ist still currently under construction and has to be cleaned. We hope that we have a first version soon.
I am currentrly working to add DimNet but I will add the AttentiveFP right after.
Thanks and best regards,
Patrick

**Comment by PatReis** [comment]
Hey, I added AttentiveFP model and an example with ESOL dataset.
Is there any particular dataset you would like to see performance of the AttentiveFP implementation?
Best
Patrick

**Comment by thegodone** [comment]
Lipo is also good to test. I see two papers HamNet and PaiNN where they predict with AttFP qm9 but it’s long tasks. Would be a nice comparison with their codes.

Envoyé de mon iPhone

**Comment by thegodone** [comment]
How fast is it ? Compare to pytorch geometric ?

Envoyé de mon iPhone

**Comment by thegodone** [comment]
I check the implementation and default parameters are not identical to source code of AttFP. hidden = 200, depth = 2. Also can you implement RMSE loss to compare with identical "target" please ?

**Comment by PatReis** [comment]
Okay, I will try to match the example with the original implementation. Is this the PyG version you mentioned in the beginning?

**Comment by thegodone** [comment]
I am looking for the original pytorch code not the PyG one: 

https://github.com/OpenDrugAI/AttentiveFP

so we can look at this one to have exact same "features". But your features are good too.

**Comment by thegodone** [comment]
https://github.com/OpenDrugAI/AttentiveFP/blob/master/code/2_Physical_Chemistry_ESOL.ipynb

**Comment by PatReis** [comment]
Okay, I tried to keep as close to the PyG and Original implementation with features and hyperparameters in latest commit.

**Comment by PatReis** [comment]
Okay, Lipop seems much better and reproducible for the paper benchmarks

**Comment by thegodone** [comment]
I agreed ESOL is split dependant not so for Lipo

Le jeu. 24 juin 2021 à 18:48, Patrick Reiser ***@***.***> a
écrit :

**Comment by PatReis** [comment]
Since these are all keras models, you can use kerastuner. I added a VERY simple example as opt_attentivefp_lipop.py

**Comment by thegodone** [comment]
perfect thankd
